config/routes.rb

```diff
@@ -1,14 +1,23 @@
 Rails.application.routes.draw do
-  resources :likes
-  resources :comments
-  resources :posts
-  resources :users
+  # resources :likes
+  # resources :comments
+  # resources :posts
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:id/posts', to: 'users#posts', as: :user_posts
+  get 'users/:id/posts/:id', to: 'users#show_user_post', as: 'show_user_post'
+  get '/users', to: 'users#index', as: 'users'
```

Comment:
- [OPTIONAL] I would recommend you to use fewer code as 
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

With this, we will mean the same as you've written